### PR TITLE
MSVC build updates

### PR DESCRIPTION
**Pull request by guruofquality**
* Removing config.h copies of M_ math constants which are provided by the addition of the compile definitions _USE_MATH_DEFINES
* Removed stdint, inttypes, and stdbool compat headers, we dont need them on recent compilers MSVC2013 and up.
* dtv: switch to hex format for number constants for msvc 2013 build
* runtime: fixed missing API export for block stream operator

### CODE REVIEW

**Comment by jmcorgan** [comment]
As this appears to be almost entirely MSVC only changes, that I can't test, I can go ahead and merge this on either master for 3.7.11 or maint for 3.7.10.2 (assuming it applies cleanly.)  Which do you prefer?

**Comment by guruofquality** [comment]
@jmcorgan If it helps, all of the patches were tested and built off of maint very recently, so they should merge. I pretty much only follow the maint branch, so I prefer maint in this case, and it lets me test the result quicker.  Of course that changes if 3.7.11 release is around the corner.

TL;DR Doesn’t bother me either way, but maint is preferred.

**Comment by jmcorgan** [comment]
Please rebase this branch and PR off maint; I will get this in for the release.

**Comment by guruofquality** [comment]
@jmcorgan I switched the branch to maint and rebased off the most recent maint (it was only a few commits behind). Thanks!
